src/_pytest/reports.py

```diff
@@ -223,7 +223,7 @@ def __init__(
         #: a (filesystempath, lineno, domaininfo) tuple indicating the
         #: actual location of a test item - it might be different from the
         #: collected one e.g. if a method is inherited from a different module.
-        self.location = location
+        self.location = location  # type: Tuple[str, Optional[int], str]
```

Comment:
I *think* that adding an annotation to a function whose own signature is unannotated doesn't have an effect ([ref](https://github.com/python/mypy/issues/3948)).

So it is better to add this annotation to the `location` parameter instead, if possible. This will make mypy consider the function to be typed.

Reply:
`reveal_type` gives me `'Tuple[builtins.str, Union[builtins.int, None], builtins.str]'` here (mypy 0.750+dev.84126ab9985844d8cda3809688034af895fdaf7c.dirty).

This appears to come / being enabled via BaseReport - adding it there instead.

Reply:
But it is needed to fix "List or tuple expected as variable arguments" with https://github.com/blueyed/pytest/blob/0133097943e0140ab82e07d4d201405145d35e59/src/_pytest/terminal.py#L442.
But it seems strange that with "location: Tuple[str, Optional[int], str]" in `TestReport.__init__` and `rep` being `_pytest.reports.TestReport`, `rep.location` is `Union[Tuple[builtins.str, Union[builtins.int, None], builtins.str], None]` then?

The problem appears to be that BaseReport has `location = None  # type: Optional[Tuple[str, Optional[int], str]]`, and we cannot have `location: Tuple[str, Optional[int], str]` in TestReport (py35).
Adding an assert appears to work also, but then again it is better to keep the type comment there.